lte/gateway/c/session_manager/SetMessageManagerHandler.cpp

```diff
@@ -86,22 +94,24 @@ void SetMessageManagerHandler::SetAmfSessionContext(
      */
     // Requested message from AMF to release the session
     if (cfg.common_context.sm_session_state() == RELEASED_4) {
-      if (cfg.common_context.sm_session_version() != 0) {
-        MLOG(MERROR) << "Wrong version received from AMF for IMSI " << imsi
-                     << " but continuing release request";
+      if (cfg.common_context.sm_session_version() == 0) {
+        MLOG(MERROR) << "Wrong version received from AMF for IMSI " << imsi;
+        Status status(grpc::OUT_OF_RANGE, "Version number Out of Range");
+        response_callback(status, SmContextVoid());
+        return;
       }
       MLOG(MINFO) << "Release request for session from IMSI: " << imsi
-                  << " DNN " << dnn;
+                  << " pdu_id " << pdu_id;
       /* Read the SessionMap from global session_store,
        * if it is not found, it will be added w.r.t imsi
        */
       auto session_map = session_store_.read_sessions({imsi});
-      initiate_release_session(session_map, dnn, imsi);
+      initiate_release_session(session_map, pdu_id, imsi);
+      response_callback(Status::OK, SmContextVoid());
     } else {
       // The Event Based main_thread invocation and runs to handle session state
-      std::string session_id = id_gen_.gen_session_id(imsi);

```

Comment:
I'm missing some context here... Do we not care about SessionID anmore in AMF?

Reply:
Yes, few changes. For an IMSI if there are multiple sessions --between AMF to SMF we use "pdu_id" to differentiate a session, earlier we use "dnn"